src/datasources/relay-api/gelato-api.service.spec.ts

```diff
@@ -0,0 +1,187 @@
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress();
+      const cacheDir = new CacheDir(`${chainId}_relay_${address}`, '');
+
+      const count = faker.number.int();
+
+      await fakeCacheService.set(cacheDir, count.toString());
+
+      const result = await target.getRelayCount({
+        chainId,
+        address,
+      });
+
+      expect(result).toEqual(count);
+    });
+
+    it('should return 0 if the cache is empty', async () => {
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress();
+
+      const result = await target.getRelayCount({
+        chainId,
+        address,
+      });
+
+      expect(result).toEqual(0);
+    });
+  });
+
+  describe('relay', () => {
+    it('should relay the payload', async () => {
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress() as Hex;
+      const data = faker.string.hexadecimal() as Hex;
+      const apiKey = faker.string.sample();
+      const taskId = faker.string.alphanumeric();
+
+      fakeConfigurationService.set(`gelato.apiKey.${chainId}`, apiKey);
+
+      mockGelatoClient.sponsoredCall.mockResolvedValue({ taskId });
+
+      const result = await target.relay({
+        chainId,
+        to: address,
+        data,
+      });
+
+      expect(result).toEqual({ taskId });
+      expect(mockGelatoClient.sponsoredCall).toHaveBeenCalledWith(
+        {
+          chainId: BigInt(chainId),
+          data,
+          target: address,
+        },
+        apiKey,
+        {
+          gasLimit: undefined,
+        },
+      );
+    });
+
+    it('should add a gas buffer if a gas limit is provided', async () => {
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress() as Hex;
+      const data = faker.string.hexadecimal() as Hex;
+      const gasLimit = faker.number.bigInt();
+      const apiKey = faker.string.sample();
+
+      fakeConfigurationService.set(`gelato.apiKey.${chainId}`, apiKey);
+
+      await target.relay({
+        chainId,
+        to: address,
+        data,
+        gasLimit,
+      });
+
+      expect(mockGelatoClient.sponsoredCall).toHaveBeenCalledWith(
+        {
+          chainId: BigInt(chainId),
+          data,
+          target: address,
+        },
+        apiKey,
+        {
+          gasLimit: gasLimit + BigInt(150_000),
+        },
+      );
+    });
+
+    it('should throw if there is no API key preset', async () => {
+      const chainId = faker.string.numeric();
+      const address = faker.finance.ethereumAddress() as Hex;
+      const data = faker.string.hexadecimal() as Hex;
+
+      await expect(
+        target.relay({
+          chainId,
+          to: address,
+          data,
+        }),
+      ).rejects.toThrow();
+    });
+
+    it('should increment the count after relaying', async () => {
+      const setSpy = jest.spyOn(fakeCacheService, 'set');
```

Comment:
I don't think we should mix the use of a `fake` and `spy` here. We can get the count set by the fake cache and assert the returned value.

Reply:
Changed to check the return value in 641026000723aeba2b9c4abce2179b1b7d54ecd1.